### PR TITLE
completed exercises

### PR DESCRIPTION
**Pull request by Monika39-maker**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Monika Dangol
- Your City: London
- Your Slack Name: Monika Dangol

# Homework Details

- Module: js core 1
- Week: 2

# Notes

- What did you find easy?
- everything was moderately challenging

- What did you find hard?
- everything was moderately challenging. I mostly find difficult understanding the problem

- What do you still not understand?
why it did not work when I wanted to compare the number as 0 < number < 10 🤔

- Any other notes?
-

### CODE REVIEW

File: exercises/D-logical-operators/exercise2.js [hunk 1]
```diff
@@ -5,7 +5,37 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
+function isNegative(num) {
+  if (num <= 0) {
```
Comment (SamanZahedi) [inline exercises/D-logical-operators/exercise2.js:9]: Do we need to include 0 here?

File: exercises/D-logical-operators/exercise2.js [hunk 2]
```diff
@@ -5,7 +5,37 @@
...
+    return true
+  } else {
+    return false
+  }
+}
+
+function isBetween5and10(num) {
+  if (5 < num < 10 ) {
```
Comment (SamanZahedi) [inline exercises/D-logical-operators/exercise2.js:17]: Did you test to see if your function works when combining your conditions?

File: exercises/D-logical-operators/exercise2.js [hunk 3]
```diff
@@ -5,7 +5,37 @@
...
+
+function isShortName(name) {
+  if (name.length <= 7) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function startsWithD(name) {
```
Comment (SamanZahedi) [inline exercises/D-logical-operators/exercise2.js:32]: Nicely done!

File: exercises/J-array-get-set/exercises2.js [hunk 4]
```diff
@@ -7,7 +7,9 @@
 */
 
 let numbers = [1, 2, 3]; // Don't change this array literal declaration
-
+numbers[3] = 4
+numbers[0] = 5;
```
Comment (SamanZahedi) [inline exercises/J-array-get-set/exercises2.js:11]: I don't think you'll get the expected result by adding this line.

File: mandatory/2-function-creation.js [hunk 5]
```diff
@@ -16,7 +22,12 @@ Write a function that:
 - numbers greater 100 must be replaced with 100
 */
 
-function formatPercentage(num) {}
+function formatPercentage(num) {
+  if (num > 100) {
+    num = 100
+  }
+  return `${Number(num.toFixed(2))}%`
```
Comment (SamanZahedi) [inline mandatory/2-function-creation.js:29]: great use on Number(), well done Monika, can you think of other ways?

File: mandatory/2-function-creation.js [hunk 6]
```diff
@@ -25,7 +36,15 @@ Write a function that:
 - removes any forward slashes (/) in each string
 - makes all strings all lowercase
 */
-function tidyUpStrings(arrayOfStrings) {}
+function tidyUpStrings(arrayOfStrings) {
+  for (let i=0; i<arrayOfStrings.length; i++) {
```
Comment (SamanZahedi) [inline mandatory/2-function-creation.js:40]: Well done Monika.

**Review by SamanZahedi — COMMENTED** [review]
Hi Monika, I finished the code review, very nice work.

File: extra/1-radio-stations.js [hunk 7]
```diff
@@ -25,6 +29,14 @@
  * - Return only the frequencies that are radio stations.
  */
 // `getStations` goes here
+function getStations() {
+  if (isRadioStation()) {
+    return true
+  } else {
+    return getAvailableStations()
```
Comment (beckyf0keefe) [inline extra/1-radio-stations.js:36]: This function technically passes the test, however, getAvailableStations is a test function that we shouldn't be using, it does the same thing that we are trying to write our own function for! Hopefully you understand how to do this after we went through it last week but feel free to ask any questions about it if not :)

**Pull request closed without merging**